### PR TITLE
Update @testing-library/user-event to v14

### LINKED ISSUE #23080: Update @testing-library/user-event to v14
@testing-library/user-event v14 is a [recommended upgrade](https://testing-library.com/docs/user-event/intro) (we are currently on v12) and provides richer APIs to unit test our front-end components (like mouse actions other than click)

Unfortunately there are breaking changes that result in 95 failed tests. I think much of this is because there's a new `setup()` function that needs to be called)

Screen Shot 2022-06-01 at 9 45 03 AM

### PR DESCRIPTION
**Pull request by gitstart**
Fixes [#23080](https://github.com/metabase/metabase/issues/23080)

###### Before submitting the PR, please make sure you do the following

- [x] If you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.

### Tests

- [x] Run the frontend and Cypress end-to-end tests with `yarn lint && yarn test`)
- [x] If there are changes to the backend codebase, run the backend tests with `clojure -X:dev:test`
- [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
      (unless it's a tiny documentation change).
 ### Demo

Screen Shot 2022-07-13 at 10 32 09 AM

- - -
This change is [Reviewable](https://reviewable.io/reviews/metabase/metabase/24087)

### CODE REVIEW

File: frontend/src/metabase/core/components/AccordionList/AccordionList.unit.spec.js [hunk 1]
```diff
@@ -100,8 +101,8 @@ describe("AccordionList", () => {
         />,
       );
 
-      userEvent.hover(screen.getByText("Foo"));
-      expect(await screen.findByText("popover")).toBeVisible();
+      await user.hover(screen.getByText("Foo"));
+      await waitFor(() => expect(screen.findByText("popover")));
```
Comment (alxnddr) [inline frontend/src/metabase/core/components/AccordionList/AccordionList.unit.spec.js:105]: This is not a correct check — there is no "popover" dom node, so seems like the test should be fixed

**Review by alxnddr — CHANGES_REQUESTED** [review]
@gitstart tremendous job, thanks for your contribution! I left a few comments that would be great to address before merging

**Comment by gitstart** [comment]
@alxnddr Tests have been updated, kindly take a look again

**Comment by iethree** [comment]
closing due to massive number of conflicts. feel free to re-open with conflicts resolved.

**Pull request closed without merging**